### PR TITLE
Basic controller audio

### PR DESCRIPTION
**Pull request by julia-jpg**
create a endpoint to controller audio

### CODE REVIEW

File: src/main/java/org/fundacionjala/converter/controller/AudioController.java [hunk 1]
```diff
@@ -44,5 +54,53 @@ public String convertAudio(final RequestAudioParameter requestAudioParameter) th
...
+           audioParameter.setSampleRate(requestAudioParameter.getSampleRate());
+           audioParameter.setStart(requestAudioParameter.getStart());
+           audioParameter.setDuration(requestAudioParameter.getDuration());
+           audioParameter.setCut(requestAudioParameter.isCut());
+           Executor executor = new Executor();
+           ICommand audioModel = new AudioModel();
+           executor.executeList(audioModel.createCommand(audioParameter));
+           return "the file already exists";
+       }
+   }
```
Comment (samuelloza) [inline src/main/java/org/fundacionjala/converter/controller/AudioController.java]: I think what it is code repetitive

**Review by samuelloza — CHANGES_REQUESTED** [review]

File: src/main/java/org/fundacionjala/converter/controller/AudioController.java [hunk 2]
```diff
@@ -25,24 +32,68 @@
...
+       if (requestAudioParameter.getFile() == null || requestAudioParameter.getFile().isEmpty()) {
+           return "Select a file";
+       }
+       ChecksumMD5 checksumMD5 = new ChecksumMD5();
+       String checksum = "";
+       String filePath = fileUploadService.saveInputFile(requestAudioParameter.getFile());
+       checksum = checksumMD5.getMD5(filePath);
+       if (fileService.getFileByMd5(checksum) == null) {
+           fileService.saveFile(new File(filePath, checksum));
+
```
Comment (maryangela) [inline src/main/java/org/fundacionjala/converter/controller/AudioController.java:58]: please delete this blank line

File: src/main/java/org/fundacionjala/converter/executor/Executor.java [hunk 3]
```diff
@@ -34,15 +34,17 @@ public Executor() {
   }
 
   private String execute(final List<String> command) throws ExecutionException, IOException, InterruptedException {
+    System.out.println(command);
```
Comment (maryangela) [inline src/main/java/org/fundacionjala/converter/executor/Executor.java]: please delete this line

File: src/main/java/org/fundacionjala/converter/executor/Executor.java [hunk 4]
```diff
@@ -34,15 +34,17 @@ public Executor() {
...
     StringBuilder output = new StringBuilder();
+    System.out.println(output);
     BufferedReader processOutputReader = new BufferedReader(new InputStreamReader(processDuration.getInputStream()));
     String line;
     while ((line = processOutputReader.readLine()) != null) {
       output.append(line + System.lineSeparator());
     }
     processDuration.waitFor();
-    System.out.println(command.get(command.size() - 1));
+    //System.out.println(command.get(command.size() - 1));
```
Comment (maryangela) [inline src/main/java/org/fundacionjala/converter/executor/Executor.java]: please delete this line

**Review by samuelloza — APPROVED** [review]

File: src/main/java/org/fundacionjala/converter/controller/AudioController.java [hunk 5]
```diff
@@ -25,24 +32,68 @@
...
+     *
+     * @param modelParameter
+     * @param requestAudioParameter
+     * @param filePath
+     * @throws IOException
+     */
+    private void setAudioParameterValues(final ModelParameter modelParameter, final RequestAudioParameter requestAudioParameter, final String filePath) throws IOException {
+        boolean cut = true;
+        modelParameter.setInputFile(filePath);
+        modelParameter.setOutputFile("storage/convertedFiles");
```
Comment (MirkoFer-622) [inline src/main/java/org/fundacionjala/converter/controller/AudioController.java]: You should call this path instancing a CofigPath object, plese not hard-coding it.

File: src/main/java/org/fundacionjala/converter/controller/request/RequestAudioParameter.java [hunk 6]
```diff
@@ -10,33 +10,87 @@
...
-                                 final String sampleRate, final String audioCodec, final String bitRate, final int channels) {
-        super(file, format, sampleRate, audioCodec);
-        this.bitRate = bitRate;
-        this.channels = channels;
+    private String channel;
+    private String sampleRate;
+    private String start;
+    private String duration;
+    private String volume;
+    private  boolean isMetadata = false;
```
Comment (MirkoFer-622) [inline src/main/java/org/fundacionjala/converter/controller/request/RequestAudioParameter.java:25]: Please, correct blank spaces.

File: src/main/java/org/fundacionjala/converter/executor/Executor.java [hunk 7]
```diff
@@ -34,15 +34,17 @@ public Executor() {
   }
 
   private String execute(final List<String> command) throws ExecutionException, IOException, InterruptedException {
+    System.out.println(command);
     Process processDuration = new ProcessBuilder(command).redirectErrorStream(true).start();
     StringBuilder output = new StringBuilder();
+    System.out.println(output);
```
Comment (MirkoFer-622) [inline src/main/java/org/fundacionjala/converter/executor/Executor.java]: Please, avoid to push debugging code lines.

**Review by MirkoFer-622 — COMMENTED** [review]
Please, could you review my comments?

**Review by maryangela — APPROVED** [review]

**Pull request merged**